### PR TITLE
feature: add stdin path to sched_process_exec

### PR DESCRIPTION
**Pull request by roikol**
add stdin_path argument to sched_process_exec event

## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

add `stdin_path` argument to the `sched_process_exec` event.

Fixes: #2215

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

locally.

Reproduce the test by running:

- command 01: `./dist/tracee-ebpf -t e=sched_process_exec`

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: pkg/ebpf/c/tracee.bpf.c [hunk 1]
```diff
@@ -915,12 +915,12 @@ BPF_ARRAY(file_filter, path_filter_t, 3);                          // filter vfs
...
 BPF_PROG_ARRAY(prog_array, MAX_TAIL_CALL);                         // store programs for tail calls
 BPF_PROG_ARRAY(prog_array_tp, MAX_TAIL_CALL);                      // store programs for tail calls
-BPF_PROG_ARRAY(sys_enter_tails, MAX_EVENT_ID);                     // store syscall specific programs for tail calls from sys_enter
-BPF_PROG_ARRAY(sys_exit_tails, MAX_EVENT_ID);                      // store syscall specific programs for tail calls from sys_exit
-BPF_PROG_ARRAY(sys_enter_submit_tail, MAX_EVENT_ID);               // store program for submitting syscalls from sys_enter
-BPF_PROG_ARRAY(sys_exit_submit_tail, MAX_EVENT_ID);                // store program for submitting syscalls from sys_exit
-BPF_PROG_ARRAY(sys_enter_init_tail, MAX_EVENT_ID);                 // store program for performing syscall tracking logic in sys_enter
-BPF_PROG_ARRAY(sys_exit_init_tail, MAX_EVENT_ID);                  // store program for performing syscall tracking logic in sys_exits
+BPF_PROG_ARRAY(sys_enter_tails, 10240);                     // store syscall specific programs for tail calls from sys_enter
```
Comment (rafaeldtinoco) [inline pkg/ebpf/c/tracee.bpf.c:918]: These are missing alignment in comments. Also, why do you think MAX_EVENT_ID is wrong here ? This is a tail call map and we could, at most, have 1 tail call, in each map, for each existing event, no ?

File: pkg/ebpf/c/tracee.bpf.c [hunk 2]
```diff
@@ -2951,8 +2951,8 @@ int sys_enter_submit(struct bpf_raw_tracepoint_args *ctx)
 
         if (file) {
             fd_arg_task_t fd_arg_task = {
-                .pid = data.context.task.pid,
-                .tid = data.context.task.tid,
+                .pid = data.context.task.host_pid,
```
Comment (rafaeldtinoco) [inline pkg/ebpf/c/tracee.bpf.c:2954]: Is this a fix ? If it is, it should be in a single commit as it is unrelated to your change (adding an argument to an event).

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
Per above comments.

File: pkg/ebpf/c/tracee.bpf.c (hunk 1 above)
Comment (roikol) [reply pkg/ebpf/c/tracee.bpf.c:918]: sorry, this is unrelated to this PR, i'll revert back.
its just a quick fix of this bug #2202 so i could test something.

File: pkg/ebpf/c/tracee.bpf.c (hunk 2 above)
Comment (roikol) [reply pkg/ebpf/c/tracee.bpf.c:2954]: your'e right. again, not related to this PR.
i should open an issue on this one

**Comment by roikol** [comment]
> Per comments above.

thanks @rafaeldtinoco !

sorry for pushing some unrelated things here. i'll update the PR soon.

**Review by rafaeldtinoco — APPROVED** [review]
LGTM. Thanks for addressing those Roi! Sorry for the delay in merging.
